README.md

```diff
@@ -162,6 +162,22 @@ The factory will *cache the created schemas* so that
 subsequent requests for a certain schema will be super fast. Therefore you should store your schema factory in a variable, 
 but you don't need to store the individual schemas.
 
+### Match order with multiple case classes
+
+Creating multiple case classes with different combinations of annotations may lead to situations, where the match is not what is expected. The following examples assume, that the case classes extend the same trait.
+
+#### 1. No match via annotations, but a case class exists with no annotations
```

Comment:
I think the most unclear situation, i.e.

"Matches one case class with annotations and one or more case classes without annotations" 

is missing from this description. In this case the case class with annotations is matched correctly.

The same now probably applies with NotWhen and OnlyWhen used together, since there is a "weight" field. I.e.

"Matches one case class with NotWhen, one or more case classes with OnlyWhen, and one or more case classes without annotations"

Also in this case probably the case class with "NotWhen" is matched.